pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/JuniperShipKitRoute.java

```diff
@@ -0,0 +1,320 @@
+package org.broadinstitute.dsm.route;
+
+import java.sql.Connection;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.easypost.exception.EasyPostException;
+import com.easypost.model.Address;
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.DSMServer;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.InstanceSettings;
+import org.broadinstitute.dsm.db.KitRequestShipping;
+import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.db.dto.settings.InstanceSettingsDto;
+import org.broadinstitute.dsm.model.KitRequest;
+import org.broadinstitute.dsm.model.KitRequestSettings;
+import org.broadinstitute.dsm.model.KitType;
+import org.broadinstitute.dsm.model.KitUploadObject;
+import org.broadinstitute.dsm.model.KitUploadResponse;
+import org.broadinstitute.dsm.model.Value;
+import org.broadinstitute.dsm.security.RequestHandler;
+import org.broadinstitute.dsm.util.DDPKitRequest;
+import org.broadinstitute.dsm.util.EasyPostUtil;
+import org.broadinstitute.dsm.util.KitUtil;
+import org.broadinstitute.dsm.util.externalshipper.ExternalShipper;
+import org.broadinstitute.lddp.handlers.util.Result;
+import org.broadinstitute.lddp.util.DeliveryAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+public class JuniperShipKitRoute extends RequestHandler {
```

Comment:
This route at this point has many assumptions and unanswered questions, basically I have moved code from other parts of DSM and changed them to fit Juniper, but kept a commented block here and there in case the assumptions are changed in the next days